src/CosmosEditorManager.ts

```
@@ -81,11 +87,32 @@ export class CosmosEditorManager implements vscode.Disposable {
         }
     }
 
+    private async loadPersistedEditor(documentUri: vscode.Uri, tree: AzureTreeDataProvider): Promise<string> {
+        const persistedEditors = this._globalState.get(this._persistedEditorsKey);
+        //Based on the documentUri, split just the appropriate key's value on '/'
+        const editorFilePath = Object.keys(persistedEditors).find((label) => path.relative(documentUri.fsPath, label) === '');
+        if (editorFilePath) {
+            const editorNode: IAzureNode | undefined = await tree.findNode(persistedEditors[editorFilePath]);
+            let editor: ICosmosEditor;
+            if (editorNode) {
```

This should probably throw an error if editorNode is not defined